```
error[E0425]: cannot find value `browser` in this scope
   --> main.rs:186:5
    |
186 |     browser.browser.deinit();
    |     ^^^^^^^ not found in this scope

error: aborting due to previous error
```

--- a/components/compositing/compositor.rs
+++ b/components/compositing/compositor.rs
@@ -412,6 +412,10 @@ impl<Window: WindowMethods> IOCompositor<Window> {
         compositor
     }
 
+    pub fn deinit(self) {
+        self.webrender.deinit();
+    }
+
     fn start_shutting_down(&mut self) {
         debug!("Compositor sending Exit message to Constellation");
         if let Err(e) = self.constellation_chan.send(ConstellationMsg::Exit) {
@@ -1615,7 +1619,6 @@ impl<Window: WindowMethods> IOCompositor<Window> {
         1.0
     }
 }
-
 
 /// Why we performed a composite. This is used for debugging.
 #[derive(Copy, Clone, PartialEq, Debug)]

--- a/components/servo/lib.rs
+++ b/components/servo/lib.rs
@@ -268,6 +268,10 @@ impl<Window> Servo<Window> where Window: WindowMethods + 'static {
             Box::new(logger)
         }).expect("Failed to set logger.")
     }
+
+    pub fn deinit(self) {
+        self.compositor.deinit();
+    }
 }
 
 fn create_compositor_channel(event_loop_waker: Box<compositor_thread::EventLoopWaker>)

--- a/ports/servo/main.rs
+++ b/ports/servo/main.rs
@@ -183,6 +183,8 @@ fn main() {
 
     unregister_glutin_resize_handler(&window);
 
+    servo_wrapper.servo.deinit();
+
     platform::deinit()
 }
 

